fix: display decimals on largest unit

HHK1 opened the pull request:
Fixes https://github.com/EvanHahn/HumanizeDuration.js/issues/150

As explained in the issue, using the library with `largest` set (with or without units specified), results in leaving the decimal part for units larger than second.

- This fixes this behaviour and adds unit tests
- I changed the way rounding was working since it leads to [inaccurate results](https://stackoverflow.com/questions/4187146/truncate-number-to-two-decimal-places-without-rounding)
- Added some tests regarding: https://github.com/EvanHahn/HumanizeDuration.js/issues/141 since it was linked to the issue above. I don't see any problems there though.


Now, some tests are breaking, specifically:

    const h = humanizer({ largest: 2 })
    assert.strictEqual(h(540360012), '6 days, 6 hours')
    assert.strictEqual(h(540360012, { largest: 3 }), '6 days, 6 hours, 6 minutes')

I haven't touched those to avoid hiding this. I think it's reasonable to have those tests failed if `round` or `maxDecimalPoints` is not provided, the only issue is that it's going to be a "breaking" change in those cases.

You can leave it as is (and fix the test ofc), and publish a major version, or only apply the remainder if `maxDecimalPoints` has been provided, what do you think ?

EvanHahn commented:
Thanks for this.

I won't have a chance to look at this for a few days, but I've written down to take a look soon.

EvanHahn commented:
Thanks for doing this. I think this makes sense to do as a breaking change, so would you mind fixing the broken tests?

HHK1 commented:
Fixed the tests in https://github.com/EvanHahn/HumanizeDuration.js/pull/171/commits/53f05719470029d9b1be3fee097758c02b45fd61

I also realised that passing Number.Infinity was now causing a crash, so I added some tests and fixed it. Behaviour is the same as before, and is now tested.

EvanHahn commented:
Thanks. I plan to take another look in the next week or so.

HHK1 commented:
Hey @EvanHahn , any luck of having this part of the next release ? Is there anything I can do to help you review this ?

EvanHahn commented:
Sorry that I've been slow to respond here. I'm trying to take a more holistic look at the library to address some bugs like this (among others).

HHK1 commented:
Hi @EvanHahn, just checking if you want me to update this, maybe target your v4.0.0 branch directly instead of master ?

EvanHahn commented:
Sorry for moving slowly here. I'll get back to you.

EvanHahn commented:
I just renamed `master` to `main` which caused this PR to be automatically closed. I didn't realize this would happen. Apologies!
